NPE on edit applications settings

Hello, I have found an NPE when I want to save modification on an application
I juste want to remove the Wine Debug flag
https://pastebin.com/pVU6P4cb
![playonlinuxaplicationbug](https://user-images.githubusercontent.com/14032531/30514475-f3c6d0a2-9b0c-11e7-95cd-c239b7dd7945.png)

System informations:
linux mint 18.2 with kde
openjdk version "1.8.0_131"
OpenJDK Runtime Environment (build 1.8.0_131-8u131-b11-2ubuntu1.16.04.3-b11)
OpenJDK 64-Bit Server VM (build 25.131-b11, mixed mode)

I get 
```
[ERROR] org.phoenicis.library.ShortcutManager (l.170) - shortcutIcon.getPath() is null
[ERROR] org.phoenicis.library.ShortcutManager (l.171) - jar:file:/home/fevrier/POL5/POL-POM-5/phoenicis-dist/target/phoenicis-dist/lib/phoenicis-library-5.0-SNAPSHOT.jar!/org/phoenicis/library/phoenicis.png
```

--- a/phoenicis-library/src/main/java/org/phoenicis/library/ShortcutManager.java
+++ b/phoenicis-library/src/main/java/org/phoenicis/library/ShortcutManager.java
@@ -89,7 +89,7 @@ public class ShortcutManager {
                     FileUtils.copyFile(file, miniatureFile);
                 }
             }
-        } catch (IOException e) {
+        } catch (IOException | IllegalArgumentException e) {
             LOGGER.warn("Error while creating shortcut", e);
         } finally {
             libraryManager.refresh();

--- a/phoenicis-library/src/main/java/org/phoenicis/library/ShortcutManager.java
+++ b/phoenicis-library/src/main/java/org/phoenicis/library/ShortcutManager.java
@@ -165,7 +165,7 @@ public class ShortcutManager {
         final File iconBackup = new File(shortcutDirectory, baseName + ".icon_backup");
         final URI shortcutIcon = shortcutDTO.getIcon();
 
-        if (shortcutIcon != null) {
+        if (shortcutIcon != null && shortcutIcon.getPath() != null) {
             final boolean keepIcon = shortcutIcon.getPath().equals(iconFile.getPath());
             if (keepIcon) {
                 try {
@@ -182,7 +182,7 @@ public class ShortcutManager {
         final File miniatureBackup = new File(shortcutDirectory, baseName + ".miniature_backup");
         final URI shortcutMiniature = shortcutDTO.getMiniature();
 
-        if (shortcutMiniature != null) {
+        if (shortcutMiniature != null && shortcutMiniature.getPath() != null) {
             final boolean keepMiniature = shortcutMiniature.getPath().equals(miniatureFile.getPath());
             if (keepMiniature) {
                 try {

--- a/phoenicis-library/src/main/java/org/phoenicis/library/ShortcutManager.java
+++ b/phoenicis-library/src/main/java/org/phoenicis/library/ShortcutManager.java
@@ -77,19 +77,27 @@ public class ShortcutManager {
 
             FileUtils.writeStringToFile(scriptFile, shortcutDTO.getScript(), ENCODING);
 
-            if (shortcutDTO.getIcon() != null) {
-                File file = new File(shortcutDTO.getIcon());
-                if (file.exists()) {
-                    FileUtils.copyFile(file, iconFile);
+            try {
+                if (shortcutDTO.getIcon() != null) {
+                    File file = new File(shortcutDTO.getIcon());
+                    if (file.exists()) {
+                        FileUtils.copyFile(file, iconFile);
+                    }
                 }
+            } catch (IOException | IllegalArgumentException e) {
+                LOGGER.warn("Error while creating shortcut icon", e);
             }
-            if (shortcutDTO.getMiniature() != null) {
-                File file = new File(shortcutDTO.getMiniature());
-                if (file.exists()) {
-                    FileUtils.copyFile(file, miniatureFile);
+            try {
+                if (shortcutDTO.getMiniature() != null) {
+                    File file = new File(shortcutDTO.getMiniature());
+                    if (file.exists()) {
+                        FileUtils.copyFile(file, miniatureFile);
+                    }
                 }
+            } catch (IOException | IllegalArgumentException e) {
+                LOGGER.warn("Error while creating miniature", e);
             }
-        } catch (IOException | IllegalArgumentException e) {
+        } catch (IOException e) {
             LOGGER.warn("Error while creating shortcut", e);
         } finally {
             libraryManager.refresh();

